### PR TITLE
Bestworst offense

### LINKED ISSUE #19: Best/Worst Offense

### PR DESCRIPTION
**Pull request by westonio**
## Describe your changes in detail
- Adds `.count_of_teams`, `.best_offense`, and `.worst_offense` methods and applicable tests
- Adds `.avg_goals_by_team` helper method and applicable test

## Checklist before requesting a review
- [ x ] I have performed a self-review of my code
- [ x ] I have added thorough tests
- [ x ] I AM MERGING WITH Westonio/futbol

## Anything else noteworthy? (E.g. note anything that might need refactoring or special attention later)
I think we can reduce/refactor these methods more in a later stage. They are still very bulky

### CODE REVIEW

**Review by fadwil — APPROVED** [review]
This looks good, I like the use of the helper method.

File: spec/stat_tracker_spec.rb [hunk 1]
```diff
@@ -61,15 +61,22 @@
 
   describe 'League Statistics' do
     it 'calculates count_of_teams' do
-      #code here
+      expect(@stat_tracker.count_of_teams).to eq(32)
+    end
+
+    it 'has average_goals_by_team list (hash)' do
+      expect(@stat_tracker.avg_goals_by_team).to be_a(Hash)
+      expected = {"16"=>1, "17"=>1, "3"=>1, "30"=>2, "5"=>0, "6"=>2, "8"=>1, "9"=>2}
```
Comment (topher-nullset) [inline spec/stat_tracker_spec.rb:69]: nice expected variable here!

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -72,15 +72,33 @@ def average_goals_by_season
   # League Statistics
 
   def count_of_teams
+    @teams.uniq.count
+  end
 
+  #This is a helper method for .best_offense and .worst_offense
+  def avg_goals_by_team
```
Comment (topher-nullset) [inline lib/stat_tracker.rb:79]: nice helper!

**Review by topher-nullset — APPROVED** [review]
Good work and nice helper method, could be useful for more methods!

**Pull request merged**